File: gallery/src/main/java/com/qkopy/gallery/ui/camera/CameraModule.java
/*
 * Created by @ajithvgiri on 4/12/18 8:11 PM
 * Copyright (c) 2018 . All rights reserved.
 * Last modified 4/12/18 7:05 PM
 */

package com.qkopy.gallery.ui.camera;

import android.content.Context;
import android.content.Intent;
import com.qkopy.gallery.model.Config;

public interface CameraModule {
    Intent getCameraIntent(Context context, Config config);

    void getImage(Context context, Intent intent, OnImageReadyListener imageReadyListener);
}

File: gallery/src/main/java/com/qkopy/gallery/ui/camera/CameraModule.kt
package com.qkopy.gallery.ui.camera

import android.content.Context
import android.content.Intent
import com.qkopy.gallery.model.Config

interface CameraModule {
    fun getCameraIntent(
        context: Context?,
        config: Config?
    ): Intent?

    fun getImage(
        context: Context?,
        intent: Intent?,
        imageReadyListener: OnImageReadyListener?
    )
}
